main.go

```diff
@@ -169,13 +171,13 @@ func createKubeClient(apiserver string, kubeconfig string) (clientset.Interface,
 	klog.Infof("Testing communication with server")
 	v, err := kubeClient.Discovery().ServerVersion()
 	if err != nil {
-		return nil, errors.Wrap(err, "ERROR communicating with apiserver")
+		return nil, nil, errors.Wrap(err, "ERROR communicating with apiserver")
 	}
 	klog.Infof("Running with Kubernetes cluster version: v%s.%s. git version: %s. git tree state: %s. commit: %s. platform: %s",
 		v.Major, v.Minor, v.GitVersion, v.GitTreeState, v.GitCommit, v.Platform)
 	klog.Infof("Communication with server successful")
 
-	return kubeClient, nil
+	return kubeClient, config, nil
```

Comment:
create and return both the kubeclient and vpaclient (if enabled) here I would say